cs/src/core/io/IntegerHelper.cs

```diff
@@ -15,6 +15,22 @@ internal static class IntegerHelper
         public const int MaxBytesVarInt32 = 5;
         public const int MaxBytesVarInt64 = 10;
 
+        public static int GetVarUInt16Length(ushort value)
```

Comment:
Did you consider basing these on the ranges. E.g.,

```
if (value < 0x80)
{
    return 1;
}
else if (value < 0x4000)
{
    return 2;
}
else
{
    return 3;
}
```

And so on?

Here's a list of powers of 128, if you were wondering...

```
128
16384
2097152
268435456
34359738368
4398046511104
562949953421312
72057594037927900
9223372036854780000
1180591620717410000000
```